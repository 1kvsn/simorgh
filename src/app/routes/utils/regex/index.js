import services from '#lib/config/services/loadableConfig';
import {
  getArticleRegex,
  getArticleSwRegex,
  getArticleManifestRegex,
  getSwRegex,
  getManifestRegex,
  getCpsAssetRegex,
  getLiveRadioRegex,
  getPodcastEpisodeRegex,
  getPodcastBrandRegex,
  getOnDemandRadioRegex,
  getOnDemandTvRegex,
  getErrorPageRegex,
  getLegacyAssetRegex,
  getMostReadPageRegex,
  getMostReadDataRegex,
  getMostWatchedDataRegex,
  getMostWatchedPageRegex,
  getIdxPageRegex,
  getSecondaryColumnDataRegex,
  getRecommendationsDataRegex,
  getAfricaEyeTVPageRegex,
} from './utils';

// this is a bit of a mess just now

export const allServices = Object.keys(services);
export const serviceRegex = allServices.join('|');
export const variantRegex = ['/simp', '/trad', '/cyr', '/lat'].join('|');
const articleLocalRegex = ['articles', 'erthyglau', 'sgeulachdan'].join('|');
const idRegex = 'c[a-zA-Z0-9]{10}o';

export const articlePageRegex = new RegExp(
  `^/(${serviceRegex})(${articleLocalRegex})/(${idRegex})(${variantRegex})?(.amp)?$`,
);
export const articleDataPath = `/:service(${serviceRegex})/:local(${articleLocalRegex})/:id(${idRegex}):variant(${variantRegex})?.json?`;

export const articleSwPath = getArticleSwRegex(allServices);
export const articleManifestPath = getArticleManifestRegex(allServices);

export const frontPageRegex = new RegExp(
  `^/(${serviceRegex})(${variantRegex})?(.amp)?$`,
);
export const frontPageDataPath = `/:service(${serviceRegex}):variant(${variantRegex})?.json`;

export const frontPageSwPath = getSwRegex(allServices);
export const frontPageManifestPath = getManifestRegex(allServices);

export const cpsAssetPagePath = getCpsAssetRegex(allServices);
export const cpsAssetPageDataPath = `${cpsAssetPagePath}.json`;

export const liveRadioPath = getLiveRadioRegex(allServices);
export const liveRadioDataPath = `${liveRadioPath}.json`;

export const onDemandRadioPath = getOnDemandRadioRegex(allServices);
export const onDemandRadioDataPath = `${onDemandRadioPath}.json`;

export const podcastEpisodePath = getPodcastEpisodeRegex(allServices);
export const podcastEpisodeDataPath = `${podcastEpisodePath}.json`;

export const podcastBrandPath = getPodcastBrandRegex(allServices);
export const podcastBrandDataPath = `${podcastBrandPath}.json`;

export const onDemandTvPath = getOnDemandTvRegex(allServices);
export const onDemandTvDataPath = `${onDemandTvPath}.json`;

export const errorPagePath = getErrorPageRegex(allServices);

export const legacyAssetPagePath = getLegacyAssetRegex(allServices);
export const legacyAssetPageDataPath = `${legacyAssetPagePath}.json`;

export const mostReadPagePath = getMostReadPageRegex(allServices);
export const mostReadDataRegexPath = getMostReadDataRegex(allServices);

export const mostWatchedDataPath = getMostWatchedDataRegex(allServices);
export const mostWatchedPagePath = getMostWatchedPageRegex(allServices);

export const secondaryColumnDataRegexPath =
  getSecondaryColumnDataRegex(allServices);

export const recommendationsDataRegex =
  getRecommendationsDataRegex(allServices);

export const IdxPagePath = getIdxPageRegex();
export const IdxDataPath = `${IdxPagePath}.json`;

export const africaEyeTVPagePath = getAfricaEyeTVPageRegex();
export const africaEyeTVDataPath = `${africaEyeTVPagePath}.json`;
