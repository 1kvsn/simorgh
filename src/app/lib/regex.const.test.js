import * as regexes from './regex.const';

describe('regex.const', () => {
  Object.keys(regexes).forEach(regex => {
    it(`${regex} should match snapshot`, () => {
      expect(regexes[regex]).toMatchSnapshot();
    });
  });
});
