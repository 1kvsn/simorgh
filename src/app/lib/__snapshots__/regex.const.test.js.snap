// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regex.const AMP_REGEX should match snapshot 1`] = `/\\\\\\.amp\\$/`;

exports[`regex.const TLD_REGEX should match snapshot 1`] = `/\\(\\\\\\.com\\|\\\\\\.co\\\\\\.uk\\)/g`;

exports[`regex.const default should match snapshot 1`] = `
Object {
  "AMP_REGEX": /\\\\\\.amp\\$/,
  "TLD_REGEX": /\\(\\\\\\.com\\|\\\\\\.co\\\\\\.uk\\)/g,
}
`;
